pkg/kubectl/cmd/expose.go

```diff
@@ -112,6 +115,9 @@ func RunExpose(f *cmdutil.Factory, out io.Writer, cmd *cobra.Command, args []str
 		return fmt.Errorf("multiple resources provided: %v", args)
 	}
 	info := infos[0]
+	if info.Mapping.Kind != "ReplicationController" && info.Mapping.Kind != "Service" && info.Mapping.Kind != "Pod" {
```

Comment:
Do we want this? At the very least, we could/should achieve this through the factory

cc: @smarterclayton